src/open_simplex.rs

```diff
@@ -258,3 +262,445 @@ pub fn open_simplex3<T: Float>(seed: &Seed, point: &::Point3<T>) -> T {
 
     return value * math::cast(NORM_CONSTANT_3D);
 }
+
+/// 4-dimensional [OpenSimplex Noise](http://uniblock.tumblr.com/post/97868843242/noise)
+///
+/// This is a slower but higher quality form of gradient noise than
+/// `noise::perlin4`.
+pub fn open_simplex4<T: Float>(seed: &Seed, point: &math::Point4<T>) -> T {
+    fn gradient<T: Float>(seed: &Seed, stretched_floor: &math::Point4<T>, pos: &math::Point4<T>) -> T {
+        let zero: T = math::cast(0.0);
```

Comment:
bjz's probably going to mention this if I don't, but try T::zero() insead of math::cast. Also see if you can drop the explicit type.